modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -363,6 +363,38 @@ namespace cv {
                     fused_layer_names.push_back(last_layer);
                 }
 
+                void setSlice(const std::vector<int>& input_indexes, int split_size, int split_id)
+                {
+                    cv::dnn::LayerParams slice_param;
+                    slice_param.name = "Slice-name";
+                    slice_param.type = "Slice";
+
+                    int begin[] = {0, split_size * split_id, 0, 0};
+                    cv::dnn::DictValue paramBegin = cv::dnn::DictValue::arrayInt(begin, 4);
+                    slice_param.set("begin", paramBegin);
+
+                    int end[] = {-1, begin[1] + split_size, -1, -1};
+                    cv::dnn::DictValue paramEnd = cv::dnn::DictValue::arrayInt(end, 4);
+                    slice_param.set("end", paramEnd);
+
+                    for (int i = 0; i < input_indexes.size(); ++i)
+                    {
+                        darknet::LayerParameter lp;
+                        lp.layer_name = cv::format("slice_%d_%d", layer_id, input_indexes[i]);
+                        lp.layer_type = slice_param.type;
+                        lp.layerParams = slice_param;
+                        lp.bottom_indexes.push_back(fused_layer_names.at(input_indexes[i]));
+                        net->layers.push_back(lp);
+                        fused_layer_names.push_back(lp.layer_name);
+                    }
+
+                    if (input_indexes.size() == 1)
+                    {
+                        last_layer = cv::format("slice_%d_%d", layer_id, input_indexes[0]);
+                        layer_id++;
+                    }
```

Comment:
Can you please write in comments how this layer configuration works depends on number of inputs?

Reply:
Comments have been added